lib/crystalball/map_generator/class_sources_detector.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+require 'crystalball/map_generator/helpers/path_filter'
+require 'crystalball/map_generator/class_sources_detector/hierarchy_fetcher'
+require 'crystalball/map_generator/class_sources_detector/definition_tracer'
+
+module Crystalball
+  class MapGenerator
+    # Class for files paths affected object definition
+    class ClassSourcesDetector
```

Comment:
I guess it should be `ObjectSourcesDetector` since it works with singleton classes too